### PR TITLE
feat: update project credentials preference

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #8633 

### Description:

Changes
- add UI to switch preferred credentials
- add UI to create credentials for active project

Screenshot 2024-01-26 at 14 32 26

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/NavBar/UserCredentialsSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    } = useUserWarehouseCredentials();
+    const { isInitialLoading: isLoadingProjects, data: projects } =
+        useProjects();
+    const { isLoading: isLoadingActiveProjectUuid, activeProjectUuid } =
+        useActiveProjectUuid();
+    const { data: preferredCredentials } =
+        useProjectUserWarehouseCredentialsPreference(activeProjectUuid);
+    const { mutate } = useProjectUserWarehouseCredentialsPreferenceMutation();
+
+    const activeProject = useMemo(() => {
```
Comment (joaoviana) [inline packages/frontend/src/components/NavBar/UserCredentialsSwitcher.tsx:28]: wonder if `useActiveProject` could be used here 
(non-blocking)

**Review by joaoviana — COMMENTED** [review]
Code looks good! 
once base branch is merged, I can create a Render env I think

File: packages/frontend/src/components/NavBar/UserCredentialsSwitcher.tsx (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/components/NavBar/UserCredentialsSwitcher.tsx:28]: `useActiveProject` returns a uuid (string) from localstorage. 
I'm using `useActiveProjectUuid` which covers that and more scenarios. 
I agree these hooks are confusing and none returns a project object :/

**Review by joaoviana — APPROVED** [review]
💪

**Pull request merged**